LockboxXCUITests/LockBoxScreenGraph.swift

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import MappaMundi
+import XCTest
+
+let WelcomeScreen = "WelcomeScreen"
+let LockboxMainPage = "LockboxMainPage"
+let SettingsMenu = "SettingsMenu"
+
+let FxASigninScreen = "FxASigninScreen"
+let FxCreateAccount = "FxCreateAccount"
+
+let OpenSitesInMenu = "OpenSitesInMenu"
+let LockedScreen = "LockedScreen"
+let AccountSettingsMenu = "AccountSettingsMenu"
+let AutolockSettingsMenu = "AutolockSettingsMenu"
+
+let SortEntriesMenu = "SortEntriesMenu"
+
+
+class Action {
+
+    static let FxATypeEmail = "FxATypeEmail"
+    static let FxATypePassword = "FxATypePassword"
+    static let FxATapOnSignInButton = "FxATapOnSignInButton"
+    static let FxALogInSuccessfully = "FxALogInSuccessfully"
+
+    static let OpenSettingsMenu = "OpenSettingsMenu"
+
+    static let LockNow = "LockNow"
+
+    static let DisconnectFirefoxLockbox = "DisconnectFirefoxLockbox"
+    static let DisconnectFirefoxLockboxCancel = "DisconnectFirefoxLockboxCancel"
+
+    static let ChangeEntriesOrder = "ChangeEntriesOrder"
+}
+
+@objcMembers
+class FxUserState: MMUserState {
+    required init() {
+        super.init()
+        initialScreenState = WelcomeScreen
+    }
+
+    var fxaUsername: String? = nil
+    var fxaPassword: String? = nil
+}
+
+func createScreenGraph(for test: XCTestCase, with app: XCUIApplication) -> MMScreenGraph<FxUserState> {
+    let map = MMScreenGraph(for: test, with: FxUserState.self)
+
+    let navigationControllerBackAction = {
+        app.navigationBars.element(boundBy: 0).buttons.element(boundBy: 0).tap()
+    }
+
+    let cancelBackAction = {
+        app.navigationBars["Lockbox.FxAView"].buttons["Cancel"].tap()
+    }
+
+    map.addScreenState(WelcomeScreen) { screenState in
+        // Adding temporary a sleep here because now Get Start buttons appears always when relaunching the app, bug or design??
+        sleep(3)
+        if (app.buttons["Get Started"].exists) {
```

Comment:
This will run when the graph is constructed, not when the test navigator is at the Welcome Screen; so this exists test will always be false.

Reply:
Okay, I see...